### ISSUE TITLE
Error message becomes worse when lifetime is elided via `'_`

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang/rust#86759]
Given the following code: 

```rust
fn bar(val: &'_ str) {
    let a: &'static str = val;
}
```

The current output is:

```
error[E0621]: explicit lifetime required in the type of `val`
 --> src/lib.rs:2:27
  |
2 |     let a: &'static str = val;
  |                           ^^^ lifetime `'static` required

error: aborting due to previous error
```

However, if we explicitly name the lifetime:

```rust
fn bar<'a>(val: &'a str) {
    let a: &'static str = val;
}
```

the error message is much better:

```
error[E0312]: lifetime of reference outlives lifetime of borrowed content...
 --> src/lib.rs:2:27
  |
2 |     let a: &'static str = val;
  |                           ^^^
  |
  = note: ...the reference is valid for the static lifetime...
note: ...but the borrowed content is only valid for the lifetime `'a` as defined on the function body at 1:8
 --> src/lib.rs:1:8
  |
1 | fn bar<'a>(val: &'a str) {
  |        ^^

error: aborting due to previous error
```

We should emit the same error message when the lifetime is elided (referring to it as 'this elided lifetime' or something similar).

**Comment by rukai** [comment]
This is solved by https://github.com/rust-lang/rust/pull/90667
The example now outputs:
```
error[E0312]: lifetime of reference outlives lifetime of borrowed content...
 --> foo.rs:2:27
  |
2 |     let a: &'static str = val;
  |                           ^^^
  |
  = note: ...the reference is valid for the static lifetime...
note: ...but the borrowed content is only valid for the anonymous lifetime defined here
 --> foo.rs:1:13
  |
1 | fn bar(val: &'_ str) {
  |             ^^^^^^^
```

**Comment by estebank** [comment]
The [current output](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=5cb9100838b4b88c1aeb8e4b849b0bba) matches the named case:

```
error: lifetime may not live long enough
 --> src/lib.rs:2:12
  |
1 | fn bar(val: &'_ str) {
  |             - let's call the lifetime of this reference `'1`
2 |     let a: &'static str = val;
  |            ^^^^^^^^^^^^ type annotation requires that `'1` must outlive `'static`
```
```
error: lifetime may not live long enough
 --> src/lib.rs:2:12
  |
1 | fn bar<'a>(val: &'a str) {
  |        -- lifetime `'a` defined here
2 |     let a: &'static str = val;
  |            ^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'static`
```
